### PR TITLE
1057: Close keyboard on pause from web pages

### LINKED ISSUE #1057: Close keyboard when exiting feedback form
## Steps to reproduce
1. Log in to Lockwise
2. Open menu, go to **Provide Feedback**
3. Enter text in text box
4. Exit feedback

### Expected behavior
Keyboard closes

### Actual behavior
Keyboard doesn't close, has to be manually closed

image  image

### Device & build information
* Device: Pixel 2
* Build version: 3.1.0

### PR DESCRIPTION
**Pull request by eliserichards**
Fixes #1057 

## To Do

- [ ] double check the original issue to confirm it is fully satisfied
- [ ] add testing notes and screenshots in PR description to help guide reviewers
- [ ] add unit tests
  - optional: consider adding instrumentation (integration/UI) tests
- consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- [ ] request the "UX" team perform a design review (if/when applicable)
- [ ] make sure CI builds are passing (e.g.: fix lint and other errors)
- [ ] check on the [accessibility](https://mozilla-lockwise.github.io/lockwise-android/accessibility/) of any added UI

### CODE REVIEW

File: app/src/main/java/mozilla/lockbox/view/AppWebPageFragment.kt [hunk 1]
```diff
@@ -45,13 +47,25 @@ class AppWebPageFragment : BackableFragment(), WebPageView {
 
         presenter = AppWebPagePresenter(this, url)
 
-        var view = inflater.inflate(R.layout.fragment_webview, container, false)
+        val view = inflater.inflate(R.layout.fragment_webview, container, false)
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/view/AppWebPageFragment.kt]: 😍 

You might also consider using `.also` here.

Comment (eliserichards) [reply app/src/main/java/mozilla/lockbox/view/AppWebPageFragment.kt]: 🤔 I've never seen `also` before! _mind blown gif_

**Comment by eliserichards** [comment]
Putting it in `Fragment.kt` is messing with the search view. I spent (a little too much) time today trying to force it to get the right focus and softinput to show/hide, but I don't think it's worth spending more time on this. Thoughts @jhugman?

File: app/src/main/java/mozilla/lockbox/view/NetworkErrorHelper.kt [hunk 2]
```diff
@@ -12,35 +12,35 @@ import mozilla.lockbox.log
...
-        val viewParams = child.layoutParams as ConstraintLayout.LayoutParams
-        val margin = parent.resources.getDimensionPixelSize(topMarginId)
-        viewParams.setMargins(0, margin, 0, 0)
+        val viewParams = child?.layoutParams as ConstraintLayout.LayoutParams
+        val margin = parent?.resources?.getDimensionPixelSize(topMarginId)
+        viewParams.setMargins(0, margin ?: 0, 0, 0)
         child.layoutParams = viewParams
     }
 
-    fun hideNetworkError(parent: View, child: RecyclerView) {
```
Comment (eliserichards) [inline app/src/main/java/mozilla/lockbox/view/NetworkErrorHelper.kt]: Cleaned this up a bit.. The `!!` were bothering me

Comment (eliserichards) [reply app/src/main/java/mozilla/lockbox/view/NetworkErrorHelper.kt]: (see `AppWebPageFragment.kt` ln 68-69 and ln 74-75)

File: app/src/main/java/mozilla/lockbox/view/EditItemFragment.kt [hunk 3]
```diff
@@ -172,10 +170,7 @@ class EditItemFragment : BackableFragment(), EditItemDetailView {
     }
 
     override fun closeKeyboard() {
-        val imm = context?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-        if (imm.isAcceptingText) {
-            imm.hideSoftInputFromWindow(view!!.windowToken, 0)
-        }
+        super.closeKeyboard(view)
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/view/EditItemFragment.kt]: Nit: This method does nothing now; you may remove it.

Comment (eliserichards) [reply app/src/main/java/mozilla/lockbox/view/EditItemFragment.kt]: `closeKeyboard()` needs to be called from the Presenter, which is why I kept it here :)

**Pull request merged**